sim/core/spell_mod.go

```diff
@@ -65,18 +69,23 @@ func buildMod(unit *Unit, config SpellModConfig) *SpellMod {
 		removeFn = functions.Remove
 	}
 
+	if config.ResourceType > 0 && (config.ResourceType&proto.ResourceType_ResourceTypeMana|proto.ResourceType_ResourceTypeEnergy|proto.ResourceType_ResourceTypeRage|proto.ResourceType_ResourceTypeFocus == 0) {
```

Comment:
Resource type enums are not bit masks, since they are incremented by 1 integer value rather than by 1 bit, so I don't think this check will ever fail.